Add curator-operator and install prerequisites

dystewart opened the pull request:
This PR adds resources to deploy the curator operator, in this case specifically on the nerc-ocp-prod cluster. 

the first commit is adding the curator operator prerequisites. For this I've followed the instructions at [https://curator-operator.readthedocs.io/en/latest/requirements.html](https://curator-operator.readthedocs.io/en/latest/requirements.html)

TRudenko22 reviewed (approved):
All checks look good.
The only issue was a path misconfig in:
cluster-scope/bundles/curator-operator/kustomization.yaml
resolved with #251
 
quay.io/repository/leihaochen/curator-postgresql-extension was replaced by a standard postgres:13 image. No issues or errors with the controller manager at the time of writing.

jtriley commented on curator-operator/base/deploymentconfigs/postgres/deploymentconfig.yaml:
```diff
@@ -0,0 +1,56 @@
+apiVersion: apps.openshift.io/v1
+kind: DeploymentConfig
```
Just noting we have the crunchydata postgres operator on both test and prod clusters now so this, and related manifests, could eventually be replaced using a single PostgresCluster CRD in theory.

jtriley reviewed (commented):
This PR mentions deploying to the `nerc-ocp-prod` cluster but AFAICT this is only getting applied in the `nerc-ocp-test` overlays.

dystewart committed:
Add curator operator and prerequisites to prod and test clusters

kustomize and linting fixes

Add missing koku-metrics-resources

Kustomize build error fix

jtriley reviewed (dismissed):
LGTM - looks like we'll also need a PR for a curator-operator app in the https://github.com/OCP-on-NERC/nerc-ocp-apps repo

larsks reviewed (commented):
Some minor requests for metadata fixes:

- This PR appears to do something other than what is state in the description (in particular, it is adding resources to both the -prod and -test clusters).
- The commit message should probably include the same information as the PR description (what is this for/why are we doing it/where does it come from/etc).
